moabb/datasets/base.py

```diff
@@ -2,14 +2,55 @@
 Base class for a dataset
 """
 import abc
+import logging
+
+log = logging.getLogger()
 
 
 class BaseDataset(metaclass=abc.ABCMeta):
     """Base dataset"""
 
-    def __init__(self, subjects, sessions_per_subject, events, code, interval,
-                 paradigm, doi=None):
+    def __init__(self, subjects, sessions_per_subject, events,
+                 code, interval, paradigm, task_interval=None, doi=None):
+        """
+        Parameters required for all datasets
 
+        parameters
+        ----------
+        subjects: List of int
+            List of subject number # TODO: make identifiers more general
+        
+        sessions_per_subject: int
+            Number of sessions per subject
+
+        events: dict of string: int
+            String codes for events matched with labels in the stim channel. Currently imagery codes codes can include:
+            - left_hand
+            - right_hand
+            - hands
+            - feet
+            - rest
+            - left_hand_right_foot
+            - right_hand_left_foot
+            - tongue
+            - navigation
+            - subtraction
+            - word_ass (for word association)
+
+        code: string
+            Unique identifier for dataset, used in all plots
+
+        interval: list with 2 entries
+            Interval relative to trial start for imagery
+
+        paradigm: ['p300','imagery']
+            Defines what sort of dataset this is (currently only imagery is implemented)
+        
+        task_interval: list of 2 entries or None
```

Comment:
Sorry for being picky, but this change make me inconfortable, and i'm trying to avoid complicating the API more than necessary.

I do not think we need to have a double definition of time interval. The time interval should correspond to the interval of the task i.e. the motor imagery. so i would actually replace the `interval` by your task `task_interval`. 

The actual timing of the trial does not bring us anything.
We changed the API significantly, and it is the paradigm itself that is doing the epoching, so we don't event need to expose `tmin` and `tmax` in the dataset anymore. If the user want to play with cropping the epoch, it is more natural to do it at the level of the paradigm (exactly like events or channels).

To my point of view, datasets are immuable objects, they are here to abstract the data as it has been recorded.